alacritty/src/config/monitor.rs

```diff
@@ -62,32 +64,68 @@ pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventLoopProxy<Event>) {
         parents.sort_unstable();
         parents.dedup();
 
-        let watcher = debouncer.watcher();
         // Watch all configuration file directories.
         for parent in &parents {
             if let Err(err) = watcher.watch(parent, RecursiveMode::NonRecursive) {
                 debug!("Unable to watch config directory {:?}: {}", parent, err);
             }
         }
 
+        // The current debouncing time.
+        let mut debouncing_time = None;
+
+        // The events accumulated during the debounce period.
+        let mut received_events = Vec::with_capacity(8);
+
         loop {
-            let events = match rx.recv() {
-                Ok(Ok(events)) => events,
+            // We use `recv_timeout` to debounce the events comming from the watcher and reduce
+            // the amount of config reloads.
+            let event = if let Some(debouncing_time) = debouncing_time.as_mut() {
+                let time_waited = Instant::now();
+                let event = rx.recv_timeout(*debouncing_time);
+                // Adjust the debouncing time, so we debounce at least each `DEBOUNCE_DELAY`.
+                *debouncing_time = debouncing_time.saturating_sub(time_waited.elapsed());
+                event
+            } else {
+                debouncing_time = Some(DEBOUNCE_DELAY);
```

Comment:
Wouldn't it be more accurate to just store `Instant::now() + DEBOUNCE_DELAY` here and then pass that to `recv_timeout` rather than subtracting from the debounce time? The latter will definitely not be accurate, but I feel like the former is also going to be simpler code?

Reply:
The `recv_timeout` only accept `Duration` unfortunately, there's a nightly API that accepts `Instant` that will simplify the code, but we can't use it. We subtract to block at max of `DEBOUNCE_DELAY`.

Reply:
Wouldn't it still be easier to just sub now from the deadline rather than recalculating the elapsed time every time? That just seems less efficient.

Reply:
But how do you know what to sub? You need to sub the amount you've blocked, and the amount you've blocked is the elapsed time.

Reply:
All you need to do is `deadline - now` to get the duration until an instant.

Reply:
But the `now` could be in the future and thus you panic?